controllers/toolchaincluster/toolchaincluster_controller.go

```diff
@@ -61,33 +61,5 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, err
 	}
 
-	// add toolchaincluster role label if not present
-	reqLogger.Info("adding cluster role label based on type")
-	if err := r.addToolchainClusterRoleLabelFromType(ctx, toolchainCluster); err != nil {
-		return reconcile.Result{}, err
-	}
-
 	return reconcile.Result{}, r.clusterCacheService.AddOrUpdateToolchainCluster(toolchainCluster)
 }
-
-func (r *Reconciler) addToolchainClusterRoleLabelFromType(ctx context.Context, toolchainCluster *toolchainv1alpha1.ToolchainCluster) error {
-	logger := log.FromContext(ctx)
-
-	if clusterType, found := toolchainCluster.Labels[cluster.LabelType]; !found {
-		logger.Info("cluster `type` label not found, unable to add toolchain cluster role label from type")
-		return nil
-	} else if clusterType != string(cluster.Member) {
-		logger.Info("cluster `type` is not member, skipping cluster role label setting")
-		return nil
-	}
-	clusterRoleLabel := cluster.RoleLabel(cluster.Tenant)
-	if _, exists := toolchainCluster.Labels[clusterRoleLabel]; !exists {
-		logger.Info("setting cluster role label for toolchaincluster", clusterRoleLabel, toolchainCluster.Name)
-		// We use only the label key, the value can remain empty.
-		toolchainCluster.Labels[clusterRoleLabel] = ""
-		if err := r.client.Update(ctx, toolchainCluster); err != nil {
```

Comment:
Where is this label used right now? I would guess that it's safer to stop using this label first and then stop setting this label. So we can migrate everything smoothly.

Reply:
To answer @alexeykazakov , it is being used to for setting the placement roles i think, and then we have some [e2e tests](https://github.com/codeready-toolchain/toolchain-e2e/blob/34b67fd73734470c935e9c1105290d196aa50dd2/test/e2e/toolchaincluster_test.go#L146) to make sure the placement roles are set based on the cluster type as @metlos mentioned. So in this case, only the member type is supposed to have cluster role which we are setting here.

Reply:
updated the logic to accommodate this .

Reply:
As mentioned in the other comments, this logic was there only to populate the already existing ToolchainCluster CRs when we introduced the label and we apparently forgot to remove it.
The placement roles should be set via add-cluster.sh script https://github.com/codeready-toolchain/toolchain-cicd/blob/34ea85fe21ae05c9bb1a0a2bbdc986b94c8d9e92/scripts/add-cluster.sh#L387-L400 - no need to do it here as well.